app/models/Site.java

```diff
@@ -32,6 +32,6 @@ public static Site findById(String id){
         // Craig, I've started this for you, this returns all clients.
         // You need to change this to bring back only the associated clients
         // to this site, making using of the clientIds property of this class.
-      return (List<Client>) Base.toList(Client.clientsCollection.find().sort(sortString).as(Client.class));
+      return (List<Client>) Base.toList(Client.clientsCollection.find("{site:#"}, this.siteId).sort(sortString).as(Client.class));
```

Comment:
You want to pass in the id for the current Site. So that would be this.id:
```
return (List<Client>) Base.toList(Client.clientsCollection.find("{site:#"}, this.id).sort(sortString).as(Client.class));
```

Well done.